CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+cmake_minimum_required(VERSION 3.13)
+project(WickedEngine)
+
+set(CMAKE_CXX_STANDARD 17)
+set(VULKAN_SDK_DIR "/home/jeremy/VulkanSDK/1.2.135.0/x86_64")
+set(FLAGS "-L${VULKAN_SDK_DIR}/lib -lxcb -lvulkan -pthread")
+
+file(GLOB_RECURSE WK_SOURCES "WickedEngine/*.cpp")
+file(GLOB_RECURSE WK_SOURCES_C "WickedEngine/*.c")
+file(GLOB_RECURSE WK_HEADERS "WickedEngine/*.h")
+
+set (WK_INC_DIRS "")
+foreach (_headerFile ${WK_HEADERS})
+    get_filename_component(_dir ${_headerFile} PATH)
+    list(APPEND WK_INC_DIRS ${_dir})
+endforeach()
+list(REMOVE_DUPLICATES WK_INC_DIRS)
+
+add_definitions(-DNDEBUG -DCMAKE_GENERATOR_PLATFORM=x64)
+add_library(WickedEngine ${WK_SOURCES} ${WK_SOURCES_C})
+target_include_directories(WickedEngine PRIVATE ${WK_INC_DIRS} ${VULKAN_SDK_DIR}/include /usr/include/vulkan)
```

Comment:
could also consider adding the following bit to help out when using CMake + MSVC 
```cmake
set_property(GLOBAL PROPERTY USE_FOLDERS ON) # enables project folder support on MSVC
set(AUTOGEN_TARGETS_FOLDER ON) # on MSVC puts all auto targets into a single folder - helps organize things.
# re-create the directory layout in generated MSVC projects
source_group(TREE ${CMAKE_CURRENT_SOURCE_DIR} FILES ${WK_SOURCES} ${WK_SOURCES_C} ${WK_HEADERS}) 
```

Reply:
I'm not very familiar with CMake. Once I can get the pull request through, I'm hoping that someone who knows more about CMake can work on it